### PR TITLE
Default persona bar

### PR DESCRIPTION
**Pull request by JanikOskar**
@laniszew @lmazurowski @pumanitro

### CODE REVIEW

File: src/components/MyPersona/MyPersona.tsx [hunk 1]
```diff
@@ -145,7 +145,7 @@ export const MyPersona: FC = () => {
           </TextInShare>
         </ShareQr>
       </PageWrapperSpaceBetween>
-      <NavLink to="./createpersona">
+      <NavLink to="./creation/step/1/entity/persona">
```
Comment (lmazurowski) [inline src/components/MyPersona/MyPersona.tsx]: Please use APP_ROUTES here

File: src/components/QrScanner/QrScanner.tsx [hunk 2]
```diff
@@ -8,8 +8,9 @@ import { useWorkerDecode } from './hooks/useWorkerDecode';
 import { HamburgerMenu } from 'global/Layouts/HamburgerMenu/HamburgerMenu';
 import { LoginBar } from '../LoginBar/LoginBar';
 import { useQuery } from '@apollo/react-hooks';
-import { gqlUser } from 'global/graphqls/schema';
-import { GET_USER } from 'global/graphqls/User';
+import { GET_USER } from '../../global/graphqls/User';
```
Comment (lmazurowski) [inline src/components/QrScanner/QrScanner.tsx]: use absolute path here

File: src/components/QrScanner/QrScanner.tsx [hunk 3]
```diff
@@ -8,8 +8,9 @@ import { useWorkerDecode } from './hooks/useWorkerDecode';
...
+import { GET_PERSONA, GetCardType } from '../../global/graphqls/Persona';
```
Comment (lmazurowski) [inline src/components/QrScanner/QrScanner.tsx]: use absolute path here

File: src/components/QrScanner/QrScanner.tsx [hunk 4]
```diff
@@ -8,8 +8,9 @@ import { useWorkerDecode } from './hooks/useWorkerDecode';
...
+import { gqlUser } from '../../global/graphqls/schema';
```
Comment (lmazurowski) [inline src/components/QrScanner/QrScanner.tsx]: use absolute path here

File: src/components/TermandPrivacy/InformativeClausule.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import { RegulationsCards } from './RegulationsCards';
+
+export const InformativeClause = () => {
```
Comment (lmazurowski) [inline src/components/TermandPrivacy/InformativeClausule.tsx]: Add simply return form and proper types: FC<>

File: src/components/TermandPrivacy/PrivacyAndCookies.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { RegulationsCards } from './RegulationsCards';
+
+export const PrivacyAndCookies = () => {
```
Comment (lmazurowski) [inline src/components/TermandPrivacy/PrivacyAndCookies.tsx]: Add simply return form and proper types: FC<>

File: src/components/TermandPrivacy/RegulationsCards.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+const RegulationText = styled.div`
+  margin-bottom: 16px;
+  text-decoration: underline;
+`;
+
+const ArticleText = styled.div`
+  margin-bottom: 8px;
+`;
+const TextInRegulation = styled.div`
```
Comment (lmazurowski) [inline src/components/TermandPrivacy/RegulationsCards.tsx]: one space line here

File: src/components/TermandPrivacy/RegulationsCards.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  white-space: pre-wrap;
+  margin-bottom: 8px;
+`;
+
+const NumberArticle = styled.div`
+  margin-bottom: 4px;
+`;
+
+export const RegulationsCards: FC<RegulationsProps> = ({
```
Comment (lmazurowski) [inline src/components/TermandPrivacy/RegulationsCards.tsx]: You can use () => () insted of () => { return (); }

File: src/components/TermandPrivacy/TermOfUse.tsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { RegulationsCards } from './RegulationsCards';
+
+export const TermOfUse = () => {
```
Comment (lmazurowski) [inline src/components/TermandPrivacy/TermOfUse.tsx]: Add simply return form and proper types: FC<>

File: src/global/AppRouter/AppRouter.tsx [hunk 10]
```diff
@@ -34,6 +36,10 @@ export const AppRouter: FunctionComponent = () => {
       <Route path={APP_ROUTES.MY_SPOTS} exact component={MySpots} />
       <Route path={APP_ROUTES.CHOOSE_PERSONA} exact component={ChoosePersona} />
       <Route path="/my-personas" exact component={MyPersona} />
+      <Route path="/my-spots" exact component={MySpots} />
```
Comment (lmazurowski) [inline src/global/AppRouter/AppRouter.tsx]: Please add usage of APP_ROUTES here

File: src/global/AppRouter/AppRouter.tsx [hunk 11]
```diff
@@ -34,6 +36,10 @@ export const AppRouter: FunctionComponent = () => {
...
+      <Route path="/informativeclause" exact component={InformativeClause} />
```
Comment (lmazurowski) [inline src/global/AppRouter/AppRouter.tsx]: Please add usage of APP_ROUTES here

File: src/global/AppRouter/AppRouter.tsx [hunk 12]
```diff
@@ -34,6 +36,10 @@ export const AppRouter: FunctionComponent = () => {
...
+      <Route path="/privacyandcookiespolicy" exact component={PrivacyAndCookies} />
```
Comment (lmazurowski) [inline src/global/AppRouter/AppRouter.tsx]: Please add usage of APP_ROUTES here

File: src/global/AppRouter/AppRouter.tsx [hunk 13]
```diff
@@ -34,6 +36,10 @@ export const AppRouter: FunctionComponent = () => {
...
+      <Route path="/termsofuse" exact component={TermOfUse} />
```
Comment (lmazurowski) [inline src/global/AppRouter/AppRouter.tsx]: Please add usage of APP_ROUTES here

File: src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx [hunk 14]
```diff
@@ -4,18 +4,21 @@ import styled from 'styled-components';
...
-import RightProfileSvg from 'assets/RightProfileSvg.svg';
 import LogoutSvg from 'assets/Logout.svg';
-import { NavLink, useHistory } from 'react-router-dom';
+import { Link, NavLink, useHistory } from 'react-router-dom';
 import { client, PS_TOKEN_NAME } from 'global/ApolloClient/ApolloClient';
 import { SIGN_OUT, SignOutResponse } from 'global/graphqls/SignOut';
 import { APP_ROUTES } from 'global/AppRouter/routes';
 import { DrawerMenu } from 'components/Drawer/Drawer';
 import { SearchPositionBox } from './SearchPositionBox';
+import { PersonaCircle } from '../../../components/PersonaCircle/PersonaCircle';
```
Comment (lmazurowski) [inline src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx]: use absolute path

File: src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx [hunk 15]
```diff
@@ -4,18 +4,21 @@ import styled from 'styled-components';
...
+import { EntityCard as EntityType } from '../../graphqls/schema';
```
Comment (lmazurowski) [inline src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx]: use absolute path

File: src/pages/Login/Login.tsx [hunk 16]
```diff
@@ -18,7 +17,7 @@ import { Link } from 'react-router-dom';
 import { PageWrapper } from 'components/PageWrapper/PageWrapper';
 // TODO: Remove after real integration
 import { useUserContext } from 'global/UserContext/UserContext';
-import { signInWithGoogle } from '../../helpers/signInWithGoogle';
+import { APP_ROUTES } from '../../global/AppRouter/routes';
```
Comment (lmazurowski) [inline src/pages/Login/Login.tsx]: use absolute path

File: src/pages/Login/Login.tsx [hunk 17]
```diff
@@ -169,7 +180,7 @@ export const Login: FunctionComponent = () => {
                 )}
               </StyledCard>
               <RegisterCaption>
-                Don’t have account? <Link to={APP_ROUTES.REGISTER}>Register Now</Link>
+                Don’t have account? <Link to="/register">Register Now</Link>
```
Comment (lmazurowski) [inline src/pages/Login/Login.tsx]: use APP_ROUTES here

File: src/pages/Register/Register.tsx [hunk 18]
```diff
@@ -154,7 +162,11 @@ export const Register: FC = () => {
                 <StyledErrorMessage>{apiError}</StyledErrorMessage>
                 <StyledCheckbox name="termsAccepted">
                   <Caption>
-                    I read and agree to <Link to="/terms-and-conditions">Terms & Conditions</Link>
+                    I read and agree to
+                    <LinkStyle to="/termsofuse">Terms of use,</LinkStyle>
```
Comment (lmazurowski) [inline src/pages/Register/Register.tsx]: use APP_ROUTES here

File: src/pages/Register/Register.tsx [hunk 19]
```diff
@@ -154,7 +162,11 @@ export const Register: FC = () => {
...
+                    <LinkStyle to="/privacyandcookiespolicy">privacy and cookies policy</LinkStyle>
```
Comment (lmazurowski) [inline src/pages/Register/Register.tsx]: use APP_ROUTES here

File: src/pages/Register/Register.tsx [hunk 20]
```diff
@@ -154,7 +162,11 @@ export const Register: FC = () => {
...
+                    and
+                    <LinkStyleLeft to="/informativeclause">informative clause</LinkStyleLeft>
```
Comment (lmazurowski) [inline src/pages/Register/Register.tsx]: use APP_ROUTES here

File: src/components/QrScanner/QrScanner.tsx [hunk 21]
```diff
@@ -90,10 +91,21 @@ export const QrScanner = ({
...
 
   const LoginOrHamburger = () => {
+    const { data: personaData } = useQuery<GetCardType>(GET_PERSONA, {
+      variables: { uuid: data?.user.defaultPersona },
+    });
+
     if (!data) {
       return <LoginBar isLogged={true} />;
-    } else return <HamburgerMenu isWithHamburger={true} />;
+    } else if (personaData && data) {
```
Comment (laniszew) [inline src/components/QrScanner/QrScanner.tsx]: what is `data` exactly here?

Comment (laniszew) [reply src/components/QrScanner/QrScanner.tsx]: Nevermind, this component is inside QR Scanner, it should be moved outside

File: src/components/TermandPrivacy/PrivacyAndCookies.tsx (hunk 6 above)
Comment (laniszew) [reply src/components/TermandPrivacy/PrivacyAndCookies.tsx]: Just `FC` please if there is no generic type passed

File: src/components/TermandPrivacy/TermOfUse.tsx (hunk 9 above)
Comment (laniszew) [reply src/components/TermandPrivacy/TermOfUse.tsx]: Same, just `FC` will do

File: src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx [hunk 22]
```diff
@@ -4,18 +4,21 @@ import styled from 'styled-components';
...
 
 type HamburgerMenuType = {
   isWithHamburger?: boolean;
   isWithSearch?: boolean;
+  card?: EntityType | null | undefined;
```
Comment (laniszew) [inline src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx]: `undefined` is obsolete for optional prop

File: src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx [hunk 23]
```diff
@@ -89,16 +105,22 @@ export const HamburgerMenu: FC<HamburgerMenuType> = ({ isWithHamburger, isWithSe
               onClick={async () => {
                 await logout();
                 localStorage.removeItem(PS_TOKEN_NAME);
-                client.cache.reset();
                 history.push(`.${APP_ROUTES.LOGIN}`);
+                client.cache.reset();
```
Comment (laniszew) [inline src/global/Layouts/HamburgerMenu/HamburgerMenu.tsx]: is this actually called after history push? 🤔

File: src/pages/Login/Login.tsx [hunk 24]
```diff
@@ -109,10 +116,14 @@ export const Login: FunctionComponent = () => {
     const data = await signIn({ variables: { idToken } });
     const token = data?.data?.loginUser.accessToken || '';
 
-    if (token) {
+    if (token && !data?.data?.loginUser.user.defaultPersona) {
       localStorage.setItem(PS_TOKEN_NAME, token);
       setUser(data?.data?.loginUser?.user || null);
       history.push(APP_ROUTES.PERSONA_CREATION_STEP_1);
+    } else if (token && data?.data?.loginUser.user.defaultPersona) {
```
Comment (laniszew) [inline src/pages/Login/Login.tsx]: maybe it is worth to extract `defaultPersona` to const as it is used in two places here

File: src/pages/Login/Login.tsx [hunk 25]
```diff
@@ -17,9 +17,10 @@ import { Link } from 'react-router-dom';
 import { PageWrapper } from 'components/PageWrapper/PageWrapper';
 // TODO: Remove after real integration
 import { useUserContext } from 'global/UserContext/UserContext';
-import { APP_ROUTES } from 'global/AppRouter/routes';
+import { signInWithGoogle } from '../../helpers/signInWithGoogle';
```
Comment (lmazurowski) [inline src/pages/Login/Login.tsx]: use absolute path

**Pull request merged**